omnigibson/utils/usd_utils.py

```diff
@@ -697,10 +697,46 @@ def reset(cls):
         cls.MODIFIED_PRIMS = set()
 
 
+class PoseAPI:
+    DIRTY = True
+    
+    @classmethod
+    def invalidate(cls):
+        cls.DIRTY = True
+        
+    @classmethod
+    def mark_valid(cls):
+        cls.DIRTY = False
+        
+    @classmethod
+    def _refresh(cls):
+        if og.sim is not None and cls.DIRTY:
+            if og.sim._physx_fabric_interface:
+                og.sim._physx_fabric_interface.update(og.sim.get_physics_dt(), og.sim.current_time)
+            else:
+                og.sim.psi.fetch_results()
+            cls.mark_valid()
```

Comment:
1. when will `_physx_fabric_interface` be None?
2. will `og.sim._physx_fabric_interface.update(og.sim.get_physics_dt(), og.sim.current_time)` cause one physics dt to elapse?